File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.py
class Solution(object):
    def removeOuterParentheses(self, s):
        """
        :type s: str
        :rtype: str
        """
        result = ""
        opened = 0

        for char in s:
            if char == "(":
                opened += 1

                if opened > 1:
                    result += "("

            elif char == ")":
                opened -= 1

                if opened > 0:
                    result += ")"

        return result

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var removeOuterParentheses = function(s) {
    let result = "";
    let opened = 0;

    for(let char of s){
        if(char === "("){
            opened += 1;
            if(opened>1){
                result += "(";
            }
        }
        else if(char === ")"){
            opened -= 1;
            if(opened>0){
                result += ")";
            }
        }
    }

    return result;
};